README.md

```diff
@@ -1,2 +1,81 @@
-# jarvis
+# JARVIS
 Machine auto healer!
+
+
+
+## Problem
+
+
+
+For a kubernetes cluster to remain in a healthy state, all the nodes should remain in a healthy, running state.
+
+
+## Solution
+
+
+
+- Machine auto healer operator will always try to keep the nodes (machines)
+  in your cluster in a healthy, running state.
+- It will perform periodic checks on the health state of each node (machine) in your cluster.
+- If a node (machine) fails consecutive health checks over an extended time period,
+  it will initiate a repair process for that node (machine).
+
+
+
+![](./docs/images/machine_auto_healer.png)
+
+##### Node Conditions
+- The conditions field describes the status of all Running nodes.
+- By describing any node we can see the NodeCondition and its respective status
+  ```bash
+  Conditions:
+  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
+  ----             ------  -----------------                 ------------------                ------                       -------
+  MemoryPressure   False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasSufficientMemory   kubelet has sufficient memory available
+  DiskPressure     False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasNoDiskPressure     kubelet has no disk pressure
+  PIDPressure      False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasSufficientPID      kubelet has sufficient PID available
+  Ready            True    Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:45:45 +0530   KubeletReady                 kubelet is posting ready status
+
+  ```
+- Following are the NodeConditions by default supported by K8s cluster.
+  
+  
+  | ConditionType      | Condition Status   |Effect        | Key      |
+  | ------------------ | ------------------ | ------------ | -------- |
+  |Ready               |True                | -            | |
+  |                    |False               | NoExecute    | node.kubernetes.io/not-ready           |
+  |                    |Unknown             | NoExecute    | node.kubernetes.io/unreachable         |
+  |OutOfDisk           |True                | NoSchedule   | node.kubernetes.io/out-of-disk         |
```

Comment:
How to process `NoSchedule` tainted node?
Maybe set the threshold for status duration and drain the node also?